### PR TITLE
Sprint review feature

### PR DESCRIPTION
**Pull request by guiman**
Allow the generations of a Sprint Review document

Also as part of this PR there are a couple refactors:
* Replaced Singleton Configuration classes into Repository class that concentrates configurations.
* Make Configuration app play ball with new Repository implementation.
* Extract minutes feature into own app

### CODE REVIEW

File: pivotal/lib/operations.rb [hunk 1]
```diff
@@ -21,6 +21,13 @@ def stories(filter)
       end
     end
 
+    def iteration(number)
+      uri = URI.parse("https://www.pivotaltracker.com/services/v5/projects/#{@configuration.project}/iterations?offset=#{number}")
+      uri.open('X-TrackerToken' => @configuration.token) do |f|
+        JSON.parse(f.read)
```
Comment (dwfait) [inline pivotal/lib/operations.rb]: You could probably do with an abstraction here that you can send an endpoint like '/iterations?offset=#{number}' and a block to take the handle of the uri.open

File: app.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@
                                    mailgun_configuration: Pivotal::Configuration::Mailgun.instance)
 Pivotal::Configuration::Minutes.set!(usual_attendees: ENV["MINUTES_USUAL_ATTENDEES"],
                                      email: Pivotal::Configuration::Email.instance)
+Pivotal::Configuration::SprintReview.set!(team_members: ENV["SPRINT_REVIEW_TEAM_MEMBERS"])
```
Comment (dwfait) [inline app.rb]: Stuff like this could go in an initializer of sorts which is merely required, to help this file get less cluttered

File: app.rb [hunk 3]
```diff
@@ -31,6 +32,15 @@ def create_minutes_data(allow_email_minutes: false)
...
     end
+
+    def create_sprint_review_data
+      Pivotal::SprintReview.new(remaining_weeks: params['remaining_weeks'].to_i,
+                                iteration: params['iteration'],
+                                team_members: params['team_members'].split(','),
+                                change_from_planning: params['change_from_planning'].split(','),
+                                retro_actions: params['retro_actions'].split(','),
+                                planning_actions: params['planning_actions'].split(','))
+    end
```
Comment (dwfait) [inline app.rb]: Again, I'd extract this out to another place to help this file from getting cluttered (sinatra apps can very quickly become 'god classes'). Think of it as just a controller - should just be directing flow.

File: pivotal/lib/configuration/sprint_review.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  module Configuration
+    class SprintReview
+      attr_reader :team_members
+
+      def initialize(team_members: raise)
+        @team_members = team_members
+      end
+
+      def self.instance
+        @@instance
```
Comment (dwfait) [inline pivotal/lib/configuration/sprint_review.rb]: Discussed already - but singletons are bad m'kay (global state)

File: pivotal/lib/sprint_review.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    def worked_hours
+      999
+    end
+
+    def points_completed
+      @iterations.first.fetch("stories").inject(0) { |sum, story|  sum += story["estimate"]}
+    end
+
+    def choir_and_bugs
```
Comment (dwfait) [inline pivotal/lib/sprint_review.rb]: chore*

**Comment by guiman** [comment]
Update:

* Removed cluther on controller and delegated responsibility to each feature class

**Comment by guiman** [comment]
Update:

* Moving away from singletons to a Repository class wich concentrates all configuration instances. This class would be as close as a "Database" as possible

**Comment by guiman** [comment]
Update:

* Introducing `save!` and `create!` methods for `Configuration` classes. These methods abstract the storing operations into the `Repository`.
* Add test coverage for Configuration classes

**Comment by guiman** [comment]
Update:

* Remove Pivotal::Operations singleton into simple instance

**Pull request merged**